internal/gossip/gossip.go

```diff
@@ -60,6 +60,9 @@ func (g *InMemoryGossip) Subscribe(channel string, callback func(data []byte)) e
 
 func (g *InMemoryGossip) Start() error {
 	g.channel = make(chan []byte, 10)
+	g.eg = &errgroup.Group{}
```

Comment:
if we're not using context or limit, why bother with errgroup? Is there some other benefit?

Reply:
I changed to wait group. The reason why I'm using a control group for goroutines is because I want to make sure we are not leaking anything

Reply:
OK; for what it's worth, I had missed that benefit of errgroup and would have been happy for you to leave it. But the waitgroup is good too.

Reply:
I was able to add it back cuz I had to do a rebase against the feature branch anyway 🥳